README.md

```diff
@@ -263,9 +263,9 @@ additional information._
 
 ### 5.2 Delegate Design
 
-The `DelegateInterface` defines a single method that accepts a request and
-returns a response. The delegate interface must be implemented by any middleware
-dispatcher that uses middleware implementing `MiddlewareInterface`.
+A delegates is a representative of the middleware dispatcher, whose only purpose
+is to encapsulate the remaining request processing. The `DelegateInterface` defines a
+`process` method therefor, which accepts a request and returns a response.
```

Comment:
This change provides some insight, but the language is totally different than the "Middleware Design" section. Perhaps something more like:

> The `DelegateInterface ` defines a single method that accepts a server request and must return a response. This is done by dispatching the next available middleware, or if none is available, by creating an empty response.